ros_ign_bridge/test/serverclient/ign_server.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <ignition/msgs/boolean.pb.h>
+#include <ignition/msgs/world_control.pb.h>
+
+#include <ignition/transport.hh>
+
+#include <atomic>
+#include <chrono>
+#include <csignal>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include "utils/test_utils.hpp"
+#include "utils/ign_test_msg.hpp"
+
+/// \brief Flag used to break the publisher loop and terminate the program.
+static std::atomic_flag g_terminateSrv(false);
+
+//////////////////////////////////////////////////
+/// \brief Function callback executed when a SIGINT or SIGTERM signals are
+/// captured. This is used to break the infinite loop that handles requests
+/// and exit the program smoothly.
+void signal_handler(int _signal)
+{
+  if (_signal == SIGINT || _signal == SIGTERM) {
+    g_terminateSrv.clear();
+  }
+}
+
+//////////////////////////////////////////////////
+bool control_world(
+  const ignition::msgs::WorldControl &,
+  ignition::msgs::Boolean & _res)
+{
+  _res.set_data(true);
+  return true;
+}
+
+//////////////////////////////////////////////////
+int main(int /*argc*/, char **/*argv*/)
+{
+  g_terminateSrv.test_and_set();
+  // Install a signal handler for SIGINT and SIGTERM.
+  std::signal(SIGINT, signal_handler);
+  std::signal(SIGTERM, signal_handler);
+
+  // Create a transport node and advertise a topic.
+  ignition::transport::Node node;
+
+  // ignition::msgs::Color.
+  node.Advertise(
+    "/ign_ros/test/serviceclient/service",
```

Comment:
It would be good to make this service specific to the request type so we can extend the test later, something like

```suggestion
    "/ign_ros/test/serviceclient/control_world",
```

Reply:
done in https://github.com/ignitionrobotics/ros_ign/pull/211/commits/1b4c9351912033f0a457042fff8b63f3ed059b22